src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Win32.cs

```diff
@@ -45,8 +45,12 @@ private unsafe bool GetData()
                 default:
                     int error = (int)Interop.NtDll.RtlNtStatusToDosError(status);
 
+                    // ERROR_FILE_NOT_FOUND or ERROR_NO_MORE_FILES can be reached when enumerating directories in a completely empty drive root (and there isn't even a System Volume Information folder).
```

Comment:
We already track `NO_MORE_FILES` above- it is not related to nothing being there, it happens for every directory when it gets to the end of the enumeration (outside of the special case for the no files).

I think it would be better to add the NT status code (`STATUS_NO_SUCH_FILE`) in a case above `default:` (which still keeps the common cases first).  That would be a bit more targeted at this known scenario and keep this condition simpler. The comment would be a bit clearer if it said something like `// STATUS_NO_SUCH_FILE can occur when there are NO files in a volume root (usually there are hidden system files)`.

I did see a few other error codes that can translate to `ERROR_FILE_NOT_FOUND` (by blasting through error codes), but I don't think any of them should be expected here.